config.yaml

```diff
@@ -1,13 +1,13 @@
 options:
   port:
-    type: string
-    default: '9000'
+    type: int
```

Comment:
If this is string, we'll have to convert it to an int in the pod spec. If it's an int, we'll have to convert it to string in the relation data.

Reply:
That's the good part about using the serialized data interface. In relation data there is only a data field which will pass through all the data in a serialized way. See here for [provides](https://github.com/canonical/serialized-data-interface/blob/main/provide_interface.py#L25) and here for [requires](https://github.com/canonical/serialized-data-interface/blob/main/require_interface.py#L65d). This way we can keep typing our config options.